src/lib/storage/proposal/space_maker.rb

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2015] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "fileutils"
+require "storage"
+require "storage/planned_volume"
+require "storage/disk_size"
+require "storage/free_disk_space"
+require "storage/proposal/exceptions"
+require "storage/proposal/refined_devicegraph"
+
+module Yast
+  module Storage
+    module Proposal
+      using RefinedDevicegraph
+
+      # Class to provide free space for creating new partitions - either by
+      # reusing existing unpartitioned space, by deleting existing partitions
+      # or by resizing an existing Windows partition.
+      class SpaceMaker
+        include Yast::Logger
+
+        # Initialize.
+        #
+        # @param original_graph [::Storage::Devicegraph] initial devicegraph
+        def initialize(original_graph)
+          @original_graph = original_graph
+        end
+
+        # Returns a copy of the original devicegraph in which all needed
+        # operations to free the required size have been performed
+        #
+        # @raise Proposal::NoDiskSpaceError if there is no enough room
+        #
+        # @param required_size [DiskSize] required amount of available space
+        # @return [::Storage::Devicegraph]
+        def run(required_size)
```

Comment:
Again: "run" is too generic here. Why replace perfectly speaking names with generic ones that don't tell anything at all? It's like calling the main function "doit()". I don't like this at all. This is not an improvement.